modules/kernel/src/main/scala/trace4cats/kernel/headers/GoogleCloudTraceToHeaders.scala

```diff
@@ -28,26 +28,31 @@ private[trace4cats] object GoogleCloudTraceToHeaders {
   // from https://cloud.google.com/trace/docs/setup
   val headerPattern =
     """(?xi)
-      |([0-9a-f]+) # trace ID
+      |([^\/]+)    # trace ID
       |\/
-      |(\d+)       # span ID (unsigned decimal)
+      |([^;]+)     # span ID (unsigned decimal)
+      |(?:
       |;
-      |o=(0|1)     # trace enabled flag
+      |o=(.*)      # trace enabled flag
+      |)?
       |""".stripMargin.r
 
   def parse(header: String): Either[Throwable, SpanContext] = header match {
     case headerPattern(traceId, spanId, enabled) =>
       for {
         traceId <- Either.fromOption(TraceId.fromHexString(traceId), new Exception("invalid trace ID"))
         spanId <- Either.fromOption(
-          SpanId.fromHexString("%016x".format(BigInt(spanId))),
+          Either.catchNonFatal(BigInt(spanId)).toOption.flatMap(bi => SpanId.fromHexString("%016x".format(bi))),
           new Exception("invalid span ID")
         )
       } yield SpanContext(
         traceId = traceId,
         spanId = spanId,
         parent = none,
-        traceFlags = TraceFlags(if (enabled == "1") SampleDecision.Include else SampleDecision.Drop),
+        traceFlags = TraceFlags(Option(enabled) match { // `enabled` can be `null`
```

Comment:
The change here swaps the default for unexpected values from dropping to sampling – I think it would be better to do `case Some("1") | None => SampleDecision.Include` instead.

Reply:
Oh, just saw there was already a discussion on that – I'm rather in the explicit sampling camp ⛺

Reply:
This sounds reasonable, I switched to `case Some("1") | None => SampleDecision.Include`